ttnn/cpp/ttnn/operations/moreh/moreh_adam/moreh_adam_pybind.cpp

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_adam_pybind.hpp"
+
+#include "pybind11/decorators.hpp"
+#include "ttnn/operations/moreh/moreh_adam/moreh_adam.hpp"
+
+namespace ttnn::operations::moreh::moreh_adam {
+void bind_moreh_adam_operation(py::module& module) {
+    bind_registered_operation(
+        module,
+        ttnn::moreh_adam,
+        "Moreh Adam Operation",
+        ttnn::pybind_arguments_t{
+            py::arg("param_in"),
+            py::arg("grad"),
+            py::arg("exp_avg_in"),
+            py::arg("exp_avg_sq_in"),
+
+            py::arg("lr"),
+            py::arg("beta1"),
+            py::arg("beta2"),
+            py::arg("eps"),
+            py::arg("weight_decay"),
+            py::arg("step"),
+            py::arg("amsgrad"),
+
+            py::arg("max_exp_avg_sq_in") = std::nullopt,
+            py::arg("param_out") = std::nullopt,
+            py::arg("exp_avg_out") = std::nullopt,
+            py::arg("exp_avg_sq_out") = std::nullopt,
+            py::arg("max_exp_avg_sq_out") = std::nullopt,
+
+            py::arg("memory_config") = std::nullopt,
```

Comment:
In ttnn we usually use kw_only arguments dtype and memory_config.

Also make sure that arguments which are kw args in torch are also kw args here.

Thank you!

Reply:
I basically faithfully ported them from the old code. I think the original logic is that if there is an output that would be pre-allocated, users have to explicitly specifies those buffers to be used, likely to avoid confusion. Should I modify the behavior here?

Reply:
I've made them `kw_only`